components/engine_traits/src/util.rs

```diff
@@ -21,3 +23,30 @@ pub fn check_key_in_range(
         ))
     }
 }
+
+pub fn append_expire_ts(value: &mut Vec<u8>, expire_ts: u64) {
+    value.encode_u64(expire_ts).unwrap();
+}
+
+pub fn get_expire_ts(value_with_ttl: &[u8]) -> Result<u64> {
+    let len = value_with_ttl.len();
+    if len < number::U64_SIZE {
+        return Err(Error::Codec(codec::Error::ValueLength));
+    }
+    let mut ts = &value_with_ttl[len - number::U64_SIZE..];
+    Ok(number::decode_u64(&mut ts)?)
+}
+
+pub fn strip_expire_ts(value_with_ttl: &[u8]) -> &[u8] {
+    let len = value_with_ttl.len();
+    &value_with_ttl[..len - number::U64_SIZE]
```

Comment:
Do we need to check the length and return a `Result`?

Reply:
It was used in `value() of iter` which returns &[u8] and it is ensured to be called after `get_expire_ts()`.

Reply:
OK. But it is a dangerous  design...